### PR TITLE
Adds CommandContext to the Cmder interface

### LINKED ISSUE #1364: Add a CommandContext to the exec package
The standard library exposes the `exec.CommandContext` function that takes a context. It would be great if kind's exec library also exposed the context version of the exec.Command function.

This could be an additional function added to `LocalCmder` and either a new interface or a new function added to the `Cmder` interface.

I would be happy to open a PR for this if y'all are interested in adding this to your exec package!

**What should be cleaned up or changed**:
This should be added because there is no convenient way to pass a context into the command library. Sometimes commands can hang indefinitely and it would be really nice to add a timeout in some cases.

**Why is this needed**:
Being able to cancel long running functions in a reasonable amount of time is nice.

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

Fixes #1364

/assign @BenTheElder

### CODE REVIEW

File: pkg/build/node/internal/container/docker/exec.go [hunk 1]
```diff
@@ -83,6 +94,9 @@ func (c *containerCmd) Run() error {
 		c.args...,
 	)
 	cmd := exec.Command("docker", args...)
+	if c.ctx != nil {
```
Comment (BenTheElder) [inline pkg/build/node/internal/container/docker/exec.go]: Probably should be an if else?

Comment (chuckha) [reply pkg/build/node/internal/container/docker/exec.go]: sure, no problems!

Comment (chuckha) [reply pkg/build/node/internal/container/docker/exec.go]: how does this look?

Comment (BenTheElder) [reply pkg/build/node/internal/container/docker/exec.go]: great, thanks!

**Comment by BenTheElder** [comment]
/lgtm
/approve
thanks!
this will be super useful

**Comment by BenTheElder** [comment]
prow
/retest

**Pull request merged**